### PR TITLE
Resync to bs58 parent

### PR DESCRIPTION
**Pull request by sublimator**

### CODE REVIEW

**Comment by dcousens** [comment]
I'm happy to merge in the credits at the top,  but the rest of these are stylistic changes that do no conform to `standard`.

Please fix or close.

**Comment by sublimator** [comment]
Bugger, I ran the `standard` program and it accepted it! :) 

Lemme read up on it :)

**Comment by sublimator** [comment]
Ok, so it seemed this passed the continuous integration `standard` test, but anyway I noticed a `formatting` commit just after you added standard I'll try and reapply.

File: index.js [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
 
   function encode (buffer) {
     if (buffer.length === 0) return ''
 
-    var i, j
-    var digits = [0]
-
-    for (i = 0; i < buffer.length; i++) {
-      for (j = 0; j < digits.length; j++) digits[j] <<= 8
+    var i, j, digits = [0]
```
Comment (dcousens) [inline index.js]: This is not  standard style,  so I'm not sure how it accepted it :/

File: index.js [hunk 2]
```diff
@@ -76,9 +78,7 @@ module.exports = function base (ALPHABET) {
     }
 
     // deal with leading zeros
-    for (i = 0; string[i] === LEADER && i < string.length - 1; i++) {
-      bytes.push(0)
-    }
+    for (i = 0; string[i] === LEADER && i < string.length - 1; ++i) bytes.push(0)
```
Comment (dcousens) [inline index.js]: Oh, this is one line.  This would be OK AFAIK, but, a bit odd anyway.

File: index.js (hunk 1 above)
Comment (dcousens) [inline index.js]: @feross shouldn't this fail `standard`?

**Comment by sublimator** [comment]
What about `++i` vs `i++` in the for loops?

**Comment by dcousens** [comment]
> What about ++i vs i++ in the for loops?

Eh,  I use `++i`, others don't,  it's a non issue. 

Thanks for the PR.

File: index.js (hunk 1 above)
Comment (feross) [inline index.js]: Yes, it should. Looks like an eslint bug. I'll open an issue.

Comment (feross) [inline index.js]: Issue filed: https://github.com/eslint/eslint/issues/2786

Comment (feross) [inline index.js]: Looks like the issue was fixed in eslint. Going to publish a new version of standard now.

Comment (dcousens) [inline index.js]: Thanks @feross

**Pull request merged**